oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/binary/TextExtractionStats.java

```diff
@@ -53,25 +65,49 @@ public void log(boolean reindex) {
         }
     }
 
-    public void collectStats(ExtractedTextCache cache){
-        cache.addStats(count, totalTime, totalBytesRead, totalTextLength);
+    public long finishExtraction(long bytesRead, int extractedTextLength) {
+        long elapsedNanos = System.nanoTime() - currentExtractionStartNanos;
+        numberOfExtractions++;
+        totalBytesRead += bytesRead;
+        totalExtractedTextLength += extractedTextLength;
+        totalExtractionTimeNanos += elapsedNanos;
+        return elapsedNanos/1_000_000;
+    }
+
+    public void collectStats(ExtractedTextCache cache) {
+        cache.addStats(numberOfExtractions, totalExtractionTimeNanos/1_000_000, totalBytesRead, totalExtractedTextLength);
     }
 
     private boolean isTakingLotsOfTime() {
-        return totalTime > LOGGING_THRESHOLD;
+        return totalExtractionTimeNanos > LOGGING_THRESHOLD*1_000_000;
```

Comment:
v

Reply:
Reformatted the code to be consistent with spacing.